### ISSUE TITLE
Problem with VS code

### ISSUE THREAD

**Issue by afont1** [safing/portmaster#1447]
Hello i really like portmaster but there's a i with it when i'm using VS code or anything in it.
First there's this issue with github :
![image](https://github.com/safing/portmaster/assets/130588672/0d53d119-9786-4477-8358-b3e202f09b62)

and this issue with Cloud Code (for duet ai) :
![image](https://github.com/safing/portmaster/assets/130588672/3ae63df5-4c55-4f46-9daf-9cf488548923)

I checked VS code for blocked connections but nothing so idk can you help me
![image](https://github.com/safing/portmaster/assets/130588672/b12cbb9d-1144-4eda-9cc2-9a9077b090bc)

and of course every issues are done when i shutdown portmaster;
I'm done crying, thanks for your time and for this app, sorry for disturbing and have a great day. <3

**Comment by Raphty** [comment]
@afont1 can you pleas provide the debug info while the issue is occurring

**Comment by afont1** [comment]
@Raphty 
oh yeah of course that's before i did anything on vs code ( i just opened it) : [before.txt](https://github.com/safing/portmaster/files/14372572/before.txt)
after i did "git pull" : [after_gitpull.txt](https://github.com/safing/portmaster/files/14372595/after_gitpull.txt)
after trying to connect to cloud code : [after_cloudcode.txt](https://github.com/safing/portmaster/files/14372612/after_cloudcode.txt)

**Comment by Raphty** [comment]
@dhaavi is also taking a look on first glance I saw that you are also running avira, this might not have anything to do with the issue, but we recommend sticking with windows defender.

So if you can try deleting Avira and see if that fixes it would be my first step. 

Avira was super good and I have been using it in the past, but Defender got a lot better and is comparable to paid antivirus, running free antivirus is generally discouraged and paid ones are limited in usefulness for individuals

https://www.youtube.com/watch?v=xKrHWM1VeFQ

**Comment by Raphty** [comment]
It is still in your wfp according to the debug info.

**Comment by dhaavi** [comment]
The issue seems to be DNS.

Please check if there is anything blocked, not just within VSCode app in Portmaster.
Have you changed settings of the "System DNS Client" app?

Is your VSCode running in some kind of virtualization? VSCode has dev containers, for example. It then runs in a Docker VM. VMs are currently not supported. You can work around it be defining another DNS server within the container or the docker host / VM.

Also, Avira does not seem to be running, it might have just not uninstalled cleanly. They forgot to remove a part of the system integration. This should not be an issue though.

**Comment by anthony-rodriguez12** [comment]
## Connectivity Fix for VS Code / Copilot (Portmaster + WSL 2)
I had a very similar issue using VS Code on Windows with the WSL 2 subsystem. Even after trying various Portmaster (firewall) configurations, the final solution wasn't an application rule, but a modification to the network configuration inside the WSL environment.

The core problem was that WSL's resolv.conf was using a local nameserver (usually the Windows host IP, like 172.x.x.x or 10.x.x.x), which was either being blocked or mishandled by the firewall at the DNS level, preventing Copilot from resolving its server addresses.

🛠️ The Fix: Modify resolv.conf in WSL
The successful fix was to add the Cloudflare DNS (1.1.1.1) as the primary option in the WSL network configuration file (/etc/resolv.conf).

**Steps**:

Open your WSL terminal (Ubuntu, Debian, etc.).

Edit the resolv.conf file. You can use nano:

```sh
sudo nano /etc/resolv.conf
```
Ensure the file looks similar to the example below, placing nameserver 1.1.1.1 before any other IP (like your local Windows host IP):

```sh

# This file was automatically generated by WSL. To stop automatic generation of this file, add the following entry to /etc/wsl.conf:
# [network]
# generateResolvConf = false
nameserver 1.1.1.1
nameserver 10.255.255.254
```
Save and exit the editor.

Restart your WSL network service (or simply restart WSL/the PC).

This forces WSL to use a reliable public DNS to resolve the Copilot domains, bypassing Portmaster's implicit blocking of WSL's internal DNS resolution.
